beatport_importer.user.js

```diff
@@ -24,12 +24,23 @@ if (!unsafeWindow) unsafeWindow = window;
 $(document).ready(function () {
     MBImportStyle();
 
-    let release_url = window.location.href.replace('/?.*$/', '').replace(/#.*$/, '');
-    let release = retrieveReleaseInfo(release_url);
-    insertLink(release, release_url);
+    $.getJSON(`https://www.beatport.com/api/v4/catalog/releases/${ProductDetail.id}`).done(release_data => {
+        let release_url = window.location.href.replace('/?.*$/', '').replace(/#.*$/, '');
+        let release = retrieveReleaseInfo(release_url, release_data);
+
+        const track_urls = release_data.tracks.map(url => url.replace('api.beatport.com', 'www.beatport.com/api')).reverse();
+        let results = [];
+        const promises = track_urls.map((url, index) => $.getJSON(url).then(response => (results[index] = response)));
+
+        Promise.all(promises)
+            .then(() => results.map(result => result.isrc))
```

Comment:
```suggestion
            .then(() => results.map({ isrc, number } => ({ isrc, number }))
```
2a. Preserve the track number for the next step.